File: Forge/src/main/java/hantonik/atomic/core/item/AtomicBurnableItem.java
package hantonik.atomic.core.item;

import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.crafting.RecipeType;
import org.jetbrains.annotations.Nullable;

import java.util.function.Function;

public class AtomicBurnableItem extends AtomicItem implements IBurnableItem {
    private final int burnTime;

    public AtomicBurnableItem(int burnTime) {
        this(p -> p, burnTime);
    }

    public AtomicBurnableItem(Function<Properties, Properties> properties, int burnTime) {
        super(properties);

        this.burnTime = burnTime;
    }

    @Override
    public int getBurnTime(ItemStack stack, @Nullable RecipeType<?> type) {
        return this.burnTime;
    }
}

File: Forge/src/main/kotlin/hantonik/atomic/core/item/AtomicBurnableItem.kt
package hantonik.atomic.core.item

import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.crafting.RecipeType

open class AtomicBurnableItem(properties: (Properties) -> Properties, private val burnTime: Int) : AtomicItem(properties), IBurnableItem {
    constructor(burnTime: Int) : this({ it }, burnTime)

    override fun getBurnTime(itemStack: ItemStack, recipeType: RecipeType<*>?) = this.burnTime
}